streaming/src/main/scala/org/apache/spark/streaming/api/java/JavaStreamingContext.scala

```diff
@@ -204,9 +204,10 @@ class JavaStreamingContext(val ssc: StreamingContext) extends Closeable {
    * monitored directory by "moving" them from another location within the same
    * file system. File names starting with . are ignored.
    * @param directory HDFS directory to monitor for new file
+   * @param depth Searching depth of directory
    */
-  def textFileStream(directory: String): JavaDStream[String] = {
-    ssc.textFileStream(directory)
+  def textFileStream(directory: String, depth: Int = 1): JavaDStream[String] = {
```

Comment:
This won't work for Java API compatibility, I think. I think we need two methods, one of which has a non-optional new depth param.